app/models/course.rb

```diff
@@ -236,6 +251,14 @@ def submissions_to_review
     submissions.where('(requests_review OR requires_review) AND NOT reviewed AND NOT newer_submission_reviewed AND NOT review_dismissed')
   end
 
+  def certificate_downloadable_for?(user)
+    user.administrator? ||
+      (!user.guest? &&
+       certificate_downloadable &&
+       (certificate_unlock_spec.nil? ||
+        UnlockSpec.new(exercises.first, ActiveSupport::JSON.decode(certificate_unlock_spec)).permits_unlock_for?(user)))
```

Comment:
Maybe we could modify it to work with course (as in this case it is course level setting.) though it might be too much work..